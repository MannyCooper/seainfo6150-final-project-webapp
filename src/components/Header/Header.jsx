import React from "react";
// import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styles from "./Header.module.css";

const Header = () => {
    return (
        <header className = {styles.header}>
          <nav className = {styles.nav}>
              <Link className = {styles.nav_left} to="/">Zifan An's Blog</Link>
              <ul className = {styles.menu}>
                {/* these links should show you how to connect up a link to a specific route */}
                {/* <li>
                  <Link to="/">Posts</Link>
                </li> */}
                <li>
                  <Link to="/archives">Archives</Link>
                </li>
                <li>
                  <Link to="/tags">Tags</Link>
                </li>
                <li>
                  <Link to="/categories">Categories</Link>
                </li>
                <li>
                  <Link to="/me">Me</Link>
                </li>
              </ul>
          </nav>
      </header>
    );
};

export default Header;