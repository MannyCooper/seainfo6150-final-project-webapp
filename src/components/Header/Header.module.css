.header{
    background-color: white;
    /* background-color: rgba(255,255,255,0.7); */
    position: sticky;
    top: 0;
    z-index: 10;
}

/* @supports ((-webkit-backdrop-filter: saturate(180%) blur(20px)) or (backdrop-filter: saturate(180%) blur(20px))){
    .header{
        -webkit-backdrop-filter: saturate(180%) blur(20px);
        backdrop-filter: saturate(180%) blur(20px);
    }
} */

.nav{
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 50px;
    max-width: 1100px;
    margin: 0 auto;
}

.nav_left{
    font-size: 18px;
    font-weight: 600;
}

.menu{
    list-style: none;
}

.menu li {
    display: inline;
    margin-left: 15px;
}

.menu li:hover,.nav_left:hover {
    color: #ff375f;
}

@media only screen and (max-width: 1380px){
    .nav{
        max-width: 980px;
    }
}

@media only screen and (max-width: 1200px){
    .nav{
        width: 692px;
    }
}

@media only screen and (max-width: 768px){
    .nav{
        max-width: 366px;
        width: 87.5%;
    }
    .nav_left{
        font-size: 15px
    }
    .menu li{
        font-size: 13px;
    }
}
