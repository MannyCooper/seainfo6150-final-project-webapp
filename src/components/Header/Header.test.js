 /* global render */
 import React from "react";
 import Header from "./Header.jsx";
  
 it("renders the Footer component correctly", () => {
   const { container } = render(<Header />);
   expect(container).toMatchSnapshot();
 });
 