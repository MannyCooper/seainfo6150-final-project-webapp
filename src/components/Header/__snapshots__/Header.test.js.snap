// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Footer component correctly 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <a
        class="nav_left"
        href="/"
      >
        Zifan An's Blog
      </a>
      <ul
        class="menu"
      >
        <li>
          <a
            href="/archives"
          >
            Archives
          </a>
        </li>
        <li>
          <a
            href="/tags"
          >
            Tags
          </a>
        </li>
        <li>
          <a
            href="/categories"
          >
            Categories
          </a>
        </li>
        <li>
          <a
            href="/me"
          >
            Me
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
